src/renderer/services/midgard/poolActionsHistory.ts

```diff
@@ -71,11 +80,19 @@ export const createPoolActionsHistoryService = (
             )
           )
         )
-      )
+      ),
+      RxOp.tap((value) => {
+        if (RD.isPending(value)) {
+          setIsHistoryLoading(true)
+        } else {
+          setIsHistoryLoading(false)
+        }
+      })
```

Comment:
yeah. good question. `PoolHistoryView` is a separate View-component to avoid additional 100+ lines of independent business logic at the `PoolDetailsView` and `PoolDetailsView` actually can not know about actions loading state 'cuz [actions$](https://github.com/thorchain/asgardex-electron/blob/develop/src/renderer/services/midgard/poolActionsHistory.ts#L37) create a stream based on request parameters. but `PoolDetailsView` needs to know if data is loading to disable `Refresh` button...

Reply:
Why not just push `actions$` or a `isLoading` state down it to `PoolDetailsView`?

Reply:
Another way is to provide `const actionsIsLoading$ = actions$.pipe(RxOp.map(RD.isPending))` or something.

Reply:
> Why not just push actions$ or a isLoading state down it to PoolHistoryView?

if we do this we have to handle all request parameters at the `PoolDetailsView` directly to create an `actions$` stream (as i've mentioned before [actions$](https://github.com/thorchain/asgardex-electron/blob/develop/src/renderer/services/midgard/poolActionsHistory.ts#L37) if a function which depends on parameters)

>Another way is to provide const actionsIsLoading$ = actions$.pipe(RxOp.map(RD.isPending)) or something.

we can not do this for the same reason as `actions$` is a function. not a separate stream. seems naming is a little bit wrong...

i can create `actions$` as an observable state at the [poolActionsHistory](https://github.com/thorchain/asgardex-electron/blob/develop/src/renderer/services/midgard/poolActionsHistory.ts) but in this case we have to be careful with possible reduntant values left at this stream and reset stream's values manually when any component uses this "hot" stream.
if you say do things ^this^ way - i will do

Reply:
> if we do this we have to handle all request parameters at the PoolDetailsView directly to create an actions$ stream (as i've mentioned before actions$ if a function which depends on parameters)

It will be updated/fixed by going this way ^, but with another PR (as discussed in DM)